src/js/components/dropdown.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (C) 2018 The Trustees of Indiana University
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import keyCodes from '../utilities/keyCodes';
+
+export default class Dropdown {
+  constructor(element) {
+    this.element = element;
+    this.focusableElements = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]';
+
+    this.dropdownAttribute = '[data-dropdown]';
+
+    this.toggleAttribute = '[data-dropdown-toggle]';
+    this.toggleElement = this.element.querySelector(this.toggleAttribute);
+
+    this.menuAttribute = '[data-dropdown-menu]';
+    this.menuElement = this.element.querySelector(this.menuAttribute);
+
+    // Keeps track of the currently active dropdown and helps with focus management
+    this.activeDropdown = null;
+    this.isOpen = false;
+
+    // Bind methods
+    this._handleClick = this._handleClick.bind(this);
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    // Make sure icons don't receive focus
+    this.element.querySelector('svg').setAttribute('focusable', 'false');
+
+    this.init();
+  }
+
+  open() {
+    // Return if disabled dropdown is being opened programmatically
+    if (this.toggleElement.hasAttribute('disabled')) {
+      return;
+    }
+
+    this.isOpen = true;
+
+    this.toggleElement.setAttribute('aria-expanded', 'true');
+
+    // Remove the 'hidden' attribute to show the menu
+    this.menuElement.removeAttribute('hidden');
+
+    this.activeDropdown = this.element;
+  }
+
+  close() {
+    this.isOpen = false;
+
+    this.toggleElement.setAttribute('aria-expanded', 'false');
+
+    this.menuElement.setAttribute('hidden', '');
+  }
```

Comment:
I wonder if we should we should set `this.activeDropdown = null` here when the dropdown get's closed. I'm not sure on that. 🤔

Reply:
`this.activeDropdown` is set back to `null` within the `escape` key logic, so I think this would be a good idea!